symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV2.java

```diff
@@ -1,36 +1,196 @@
 import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
 import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.gen.api.model.AckId;
+import com.symphony.bdk.gen.api.model.V4Event;
+import com.symphony.bdk.gen.api.model.V5Datafeed;
+import com.symphony.bdk.gen.api.model.V5EventList;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import lombok.extern.slf4j.Slf4j;
 
+import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
- * A class for implementing the datafeed v1 service.
+ * A class for implementing the datafeed v2 service.
+ *
+ * This service will be started by calling {@link DatafeedServiceV2#start()}
+ *
+ * At the beginning, the BDK bot will try to retrieve the list of datafeed to which it is listening. Since each bot
+ * should only listening to just one datafeed, the first datafeed in the list will be used by the bot to be listened to.
+ * If the retrieved list is empty, the BDK bot will create a new datafeed to listen.
+ *
+ * The BDK bot will listen to this datafeed to get all the received real-time events.
+ *
+ * If this datafeed becomes stale or faulty, the BDK bot will create the new one for listening.
+ *
+ * This service will be stopped by calling {@link DatafeedServiceV2#stop()}
+ *
+ * If the datafeed service is stopped during a read datafeed call, it has to wait until the last read finish to be
+ * really stopped
  */
+@Slf4j
 public class DatafeedServiceV2 extends AbstractDatafeedService {
 
     private final AtomicBoolean started = new AtomicBoolean();
+    private final AckId ackId;
+    private V5Datafeed datafeed;
 
-    public DatafeedServiceV2(ApiClient agentClient, ApiClient podClient, AuthSession authSession, BdkConfig config) {
-        super(agentClient, podClient, authSession, config);
+    public DatafeedServiceV2(ApiClient agentClient, AuthSession authSession, BdkConfig config) {
+        super(agentClient, authSession, config);
+        this.ackId = new AckId().ackId("");
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void start() {
+    public void start() throws ApiException, AuthUnauthorizedException {
+        if (this.started.get()) {
+            throw new IllegalStateException("The datafeed service is already started");
+        }
+        try {
+            this.datafeed = this.retrieveDatafeed();
+            if (this.datafeed == null) {
+                this.datafeed = this.createDatafeed();
+            }
+            log.debug("Start reading datafeed events");
+            do {
+                this.started.set(true);
+                this.readDatafeed();
+            } while (this.started.get());
+        } catch (AuthUnauthorizedException | ApiException exception) {
+            throw exception;
+        } catch (Throwable throwable) {
+            log.error("Unknown error", throwable);
```

Comment:
Same thought as put in  symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java